pkg/apis/cyclone/v1alpha1/stage_template.go

```diff
@@ -0,0 +1,83 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/apis/core"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StageTemplate defines a template of Stage.
+type StageTemplate struct {
+	// Metadata for the resource, like kind and apiversion
+	metav1.TypeMeta `json:",inline"`
+	// Metadata for the particular object, including name, namespace, labels, etc
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Stage template specification
+	Spec StageTemplateSpec `json:"spec"`
+}
+
+// Argument defines a argument.
+type Argument struct {
+	Name    string `json:"name"`
+	Default string `json:"default"`
+}
+
+// Inputs defines stage inputs.
+type Inputs struct {
+	// Resources used as input
+	Resources []ResourceItem `json:"resource,omitempty"`
+	// Parameters used as input
+	Arguments []ArgumentItem `json:"arguments,omitempty"`
+}
+
+// Outputs defines stage output.
+type Outputs struct {
+	// Resources used as output
+	Resources []ResourceItem `json:"resource,omitempty"`
+}
+
+// StatusRef defines how to judge CRD status completion.
+type StatusRef struct {
+	// Path of field in the CRD spec that determines status of CRD workload
+	Path string `json:"path"`
+	// Value indicates CRD workload completion
+	ComplatedValue string `json:"complatedValue"`
+}
+
+// CRDSpec defines CRD workload specification.
+type CRDSpec struct {
+	// Specification of the CRD
+	YML string `json:"yaml"`
+	// How to judge CRD workload is completed
+	StatusRef StatusRef `json:"statusRef"`
+}
```

Comment:
I think we also need something like PodRef (each input output should have a reference), otherwise how do we know where to mount our input/outputs?